docs/rules/pair-to-have-been-called-assertions.md

```diff
@@ -0,0 +1,142 @@
+expect(mockFn).toHaveBeenLastCalledWith('last');
+
+// Using toBeNthCalledWith without call count
+expect(mockFn).toBeNthCalledWith(2, 'second');
+
+// Using toBeLastCalledWith without call count
+expect(mockFn).toBeLastCalledWith('last');
+```
+
+### Correct
+
+```js
+expect(mockFn).toHaveBeenCalledTimes(1);
+expect(mockFn).toHaveBeenCalledWith('arg');
+
+expect(mockFn).toBeCalledTimes(1);
+expect(mockFn).toBeCalledWith('arg');
+
+// Multiple mocks, each with call count
+expect(mockFn1).toHaveBeenCalledTimes(1);
+expect(mockFn1).toHaveBeenCalledWith('arg1');
+expect(mockFn2).toHaveBeenCalledTimes(1);
+expect(mockFn2).toHaveBeenCalledWith('arg2');
+
+// Using toHaveBeenNthCalledWith with call count
+expect(mockFn).toHaveBeenCalledTimes(2);
+expect(mockFn).toHaveBeenNthCalledWith(1, 'first');
+
+// Using toHaveBeenLastCalledWith with call count
+expect(mockFn).toHaveBeenCalledTimes(3);
+expect(mockFn).toHaveBeenLastCalledWith('last');
+
+// Using toBeNthCalledWith with call count
+expect(mockFn).toBeCalledTimes(2);
+expect(mockFn).toBeNthCalledWith(2, 'second');
+
+// Using toBeLastCalledWith with call count
+expect(mockFn).toBeCalledTimes(1);
+expect(mockFn).toBeLastCalledWith('only');
+
+// Mixed matchers with call count
+expect(mockFn).toHaveBeenCalledTimes(3);
+expect(mockFn).toHaveBeenCalledWith('arg1');
+expect(mockFn).toHaveBeenNthCalledWith(2, 'arg2');
+expect(mockFn).toHaveBeenLastCalledWith('arg3');
+
+// Empty call (no arguments) doesn't require call count
+expect(mockFn).toHaveBeenCalledWith();
+
+// Using 'not' modifier doesn't require call count
+expect(mockFn).not.toHaveBeenCalledWith('arg');
+
+// Only checking call count is fine
+expect(mockFn).toHaveBeenCalledTimes(1);
+```
+
+## Additional Checks
+
+### Contradictory Assertions
+
+This rule also detects contradictory assertions where `toHaveBeenCalledTimes(0)`
+is used together with `toHaveBeenCalledWith()`. Since `toHaveBeenCalledWith()`
+expects the mock to be called at least once, using it with
+`toHaveBeenCalledTimes(0)` is contradictory.
+
+```js
+// ❌ Incorrect - contradictory assertions
+test('foo', () => {
+  expect(mockFn).toHaveBeenCalledTimes(0);
+  expect(mockFn).toHaveBeenCalledWith('arg'); // This expects a call!
+});
+
+// ✅ Correct - consistent assertions
+test('foo', () => {
+  expect(mockFn).toHaveBeenCalledTimes(0);
+  // No CalledWith assertions
+});
+
+// ✅ Correct - consistent assertions
+test('foo', () => {
+  expect(mockFn).toHaveBeenCalledTimes(1);
+  expect(mockFn).toHaveBeenCalledWith('arg');
+});
+```
+
+## When Not To Use It
+
+If you have a specific testing strategy where checking call counts is not
+necessary or you're only interested in verifying that a function was called with
+specific arguments at least once, you can disable this rule.
+
+However, it's generally recommended to keep this rule enabled as it helps catch
+bugs where functions are called more times than expected, leading to more robust
+and reliable tests.
```

Comment:
I don't think this is particularly useful to have in this section, especially as this a rule that is enabled by default

```suggestion

```